functions/public/Invoke-WPFMicrowin.ps1

```diff
@@ -10,7 +10,19 @@ function Invoke-WPFMicrowin {
         return
     }
 
-	
+    # Ask the user where to save the file
```

Comment:
Hey @CodingWonders it seems like you are on the roll :) thank you for fixing the bugs this is great to see. Here some comments that I have. 
1. We are trying to pursue an approach of less clicking and asking fewer questions to reduce friction. "Less clicking, more doing". If possible change the UI, there is a textbox that stores ISO location called $sync.MicrowinFinalIsoLocation. What I would have done make "$env:temp\microwin.iso" a default and add a button next to this text box to set the location, so that if people don't want to set it, they don't have to.
2. Also this now looking at the code, some of my changes disappeared for some reason, I've added some code that would add some random number and timestamp to microwin.iso name. I'm pretty sure the reason why winutil hangs is because people either kill winutil during iso creation or cancel, so there is a process on the system that is still haning onto that file. If we add a random number to the file name it will solve the problem. Not sure where that change went, somehow it got overwritten.
3. There is some code left where we generate random number in functions/public/Invoke-WPFGetIso.ps1 you can use it as an example
 ``` PowerShell
    Write-Host "Setting up mount dir and scratch dirs"
    $timestamp = Get-Date -Format "yyyyMMdd_HHmmss"
    $randomNumber = Get-Random -Minimum 1 -Maximum 9999
    $randomMicrowin = "Microwin_${timestamp}_${randomNumber}"
    $randomMicrowinScratch = "MicrowinScratch_${timestamp}_${randomNumber}"
```

I was doing the same thing for the ISO but it got overwritten now :)

Reply:
For number 2, neither do I!

Reply:
I was busy with my projects, so I couldn't _really_ work on your proposals.

Now, while I have edited a bit of XAML code, I'm not good with WPF and am better at designing WinForms stuff, so I don't know how to do this.

Maybe someone else could work on that change.

Reply:
> I was busy with my projects, so I couldn't _really_ work on your proposals.
> 
> Now, while I have edited a bit of XAML code, I'm not good with WPF and am better at designing WinForms stuff, so I don't know how to do this.
> 
> Maybe someone else could work on that change.

Sounds good, please push your code through, see if Chris takes it, I think he will, and I will work on the alternative when I get a chance.

Reply:
OK, thanks!